anchore_engine/clients/services/internal.py

```diff
@@ -284,7 +284,13 @@ def dispatch(
         else:
             filtered_qry_params = None
 
-        log_body = ensure_str(body[:512]) + "..." if body and len(body) > 512 else body
+        try:
+            log_body = (
+                ensure_str(body[:512]) + "..." if body and len(body) > 512 else body
+            )
+        except UnicodeError:
+            log_body = "Unable to decode body for logging"
```

Comment:
This is prior to response adn did not want to rely on dev correctly setting the header to prevent an exception being thrown when attempting to log so I kept the try catch but put the lenght of the binary in the log message